py/server/tests/test_function_generated_table.py

```diff
@@ -39,20 +42,26 @@ def test_generated_table_1trigger(self):
         append_only_input_table = input_table(col_defs=col_defs)
 
         def table_generator_function():
+            print("Running table_generator_function() at time: " + str(datetime.now()))
             return append_only_input_table.last_by().update('ResultStr = MyStr')
 
         result_table = function_generated_table(table_generator_function, source_tables=append_only_input_table)
 
         self.assertEqual(result_table.size, 0)
 
+        print("Adding row at time: " + str(datetime.now()))
         append_only_input_table.add(new_table([string_col(name='MyStr', data=['test string'])]))
+        print("add() returned at time: " + str(datetime.now()))
         self.wait_ticking_table_update(result_table, row_count=1, timeout=30)
-        self.assertEqual(result_table.size, 1)
+        print("result_table has 1 row at time: " + str(datetime.now()))
 
         result_str = result_table.j_table.getColumnSource("ResultStr").get(0)
         self.assertEqual(result_str, 'test string')
```

Comment:
This is where we were failing, right @rbasralian ? @jmao-denver or @chipkent, is it possible that this is the wrong way to compare strings? I see some stuff in the unittest.TestCase documentation about type-specific equals that was added to assertEquals in 3.1. Is that Python 3.1, or v3.1 of the testcase package, or?

Reply:
`assertEqual` is the right way. The logged error is "None not equal to 'test string' which indicates `get(0)` returning a null value.